src/demos/trackVehicle/subsys/base/ChTrackVehicle.h

```diff
@@ -162,11 +155,36 @@ class CH_SUBSYS_API ChTrackVehicle : public ChShared {
 
     /// setup class to save the log to a file for python postprocessing.
     /// Usage: call after construction & Initialize(), else no data is saved.
-    virtual void Setup_log_to_file(int what,
-                                   const std::string& out_filename,
-                                   const std::string& data_dirname = "data_test") {}
+    virtual void Setup_logger(int what_subsys,  /// which vehicle objects (e.g. subsystems) to save data for?
+                              int debug_type,   /// data types: _BODY, _CONSTRAINTS, _CONTACTS
+                              const std::string& out_filename,
+                              const std::string& data_dirname = "data_test") {}
+
+
+    // helper functions, for Irrlicht GUI
+    // following variables are populated when DriveChain::reportShoeGearContact() is called
+    // absolute pos of the persistent contact point
+    virtual const ChVector<>& Get_SG_Persistent_PosAbs(int track, int idx) const {
+        return ChVector<>();
+    }
+
+    // normal force abs. vector of the persistent contact point
+    virtual const ChVector<>& Get_SG_Persistent_Fn(int track, int idx) const {
+        return ChVector<>(); 
+    }
+
+    // abs. pos. of all shoe-gear contacts found
+    virtual const std::vector<ChVector<> >& Get_SG_PosAbs_all(int track) const {
+        return std::vector<ChVector<>>();
```

Comment:
Should be `std::vector< ChVector<> >();` (note the spaces), at least until we default to C++11.